tracer/src/Datadog.Trace/AppSec/Waf/Waf.cs

```diff
@@ -62,7 +62,7 @@ internal static InitResult Create(WafLibraryInvoker wafLibraryInvoker, string ob
             var wafConfigurator = new WafConfigurator(wafLibraryInvoker);
 
             // set the log level and setup the logger
-            wafLibraryInvoker.SetupLogging(GlobalSettings.Instance.DebugEnabledInternal);
+            wafLibraryInvoker.SetupLogging(GlobalSettings.Instance.DebugEnabledInternal && Security.Instance.Settings.WafDebugEnabled);
```

Comment:
IMO this should be passed in as a parameter to the `Create` method, instead of reaching out to a singleton instance. This sort of thing plays havoc with testing etc

Reply:
Changed in 9876b46752